core/src/providers/openai.rs

```diff
@@ -1015,79 +1222,554 @@ pub async fn chat_completion(
+                    println!("UNEXPECTED COMMENT");
+                }
+                Some(es::SSE::Event(e)) => match e.data.as_str() {
+                    "[DONE]" => {
+                        break 'stream;
+                    }
+                    _ => {
+                        let index = {
+                            let guard = chunks.lock();
+                            guard.len()
+                        };
 
-    let res = match timeout(Duration::new(60, 0), req.send()).await {
-        Ok(Ok(res)) => res,
-        Ok(Err(e)) => Err(e)?,
-        Err(_) => Err(anyhow!("Timeout sending request to OpenAI after 60s"))?,
-    };
+                        let chunk: ChatChunk = match serde_json::from_str(e.data.as_str()) {
+                            Ok(c) => c,
+                            Err(err) => {
+                                let error: Result<Error, _> = serde_json::from_str(e.data.as_str());
+                                match error {
+                                    Ok(error) => {
+                                        match error.retryable_streamed() && index == 0 {
+                                            true => Err(ModelError {
+                                                message: error.message(),
+                                                retryable: Some(ModelErrorRetryOptions {
+                                                    sleep: Duration::from_millis(500),
+                                                    factor: 2,
+                                                    retries: 3,
+                                                }),
+                                            })?,
+                                            false => Err(ModelError {
+                                                message: error.message(),
+                                                retryable: None,
+                                            })?,
+                                        }
+                                        break 'stream;
+                                    }
+                                    Err(_) => {
+                                        Err(anyhow!(
+                                            "OpenAIError: failed parsing streamed \
+                                                 completion from OpenAI err={} data={}",
+                                            err,
+                                            e.data.as_str(),
+                                        ))?;
+                                        break 'stream;
+                                    }
+                                }
+                            }
+                        };
 
-    let body = match timeout(Duration::new(60, 0), res.bytes()).await {
-        Ok(Ok(body)) => body,
-        Ok(Err(e)) => Err(e)?,
-        Err(_) => Err(anyhow!("Timeout reading response from OpenAI after 60s"))?,
-    };
+                        // println!("CHUNK: {:?}", chunk);
```

Comment:
nit: to remove

Reply:
They were commented in the previous code but happy to remove them.